src/Microsoft.VisualStudio.Threading/JoinableTaskContext.cs

```diff
@@ -339,6 +339,27 @@ public bool IsMainThreadBlocked()
             return false;
         }
 
+        /// <summary>
+        /// Gets a proximate value whether the main thread is blocked for the caller's completion.
+        /// It is less accurate when the UI thread blocking task just starts and hasn't been blocked yet, or the dependency chain is just removed.
+        /// However, unlike <see cref="IsMainThreadBlocked"/>, this implementation is lock free, and faster in high contention scenarios.
+        /// </summary>
+        public bool IsMainThreadBlockedProximately()
```

Comment:
I think you mean `approximately`, but I've seen `Maybe` is used elsewhere for similar examples; `IsMainThreadMaybeBlocked`

Reply:
Thanks for the suggestion. I updated the code.